CO2 Data Collection of the Mauna Loa Observatory

haiderabbas007 opened the pull request:
# Pull Request: CO2 Data Collection of the Mauna Loa Observatory

## Linting

- All Python scripts in this repository are `pylint` compliant, ensuring adherence to PEP8 standards for readability, maintainability, and consistent style.

## Contents

This PR contains data files and scripts to process CO₂ concentration data collected from the Mauna Loa Observatory. The data includes various types of CO₂ measurements, such as discrete, monthly, daily, and hourly readings, stored in subdirectories corresponding to the sampling type (e.g., flask monthly, insitu daily).

- **Top-Level Files**:
  - `README.md`: Provides comprehensive instructions, directory overview, file descriptions, citation information, and sample data formats.
  - `LICENSE`: Indicates that all data files are in the public domain under the **Creative Commons Zero (CC0) 1.0 Universal Public Domain Dedication**.
  - `NOTES.md`: Additional notes for repository users.

- **Subdirectories**:
   - `flask-discrete/`
   - `flask-monthly/`
   - `insitu-daily/`
   - `insitu-hourly/`
   - `insitu-monthly/`
   - `surface-discrete/`

- **Subdirectory Files** (for each of `flask-discrete`, `flask-monthly`, `insitu-daily`, `insitu-hourly`, `insitu-monthly`, and `surface-discrete`):
  - `.csv`: Processed CO₂ data in CSV format.
  - `.json`: Processed CO₂ data in JSON format.
  - `_no_nan.json`: JSON format without NaN values.
  - `.md`: Markdown format of the CO₂ data.
  - `.py`: Python script to process the raw data file, `pylint` compliant.
  - `_README.html`: Detailed README from NOAA for data usage and license information.
  - `_raw.txt`: Original raw data file as downloaded from NOAA.

## Suggested Reviewer

@SchrodingersStruggle

haiderabbas007 commented:
All the python scripts are ```pylint``` compliant **locally** at least

SchrodingersStruggle commented:
Yeah I see that your code has a 10/10 linting score so it deserves to pass the test. @laserlab do you think it's worth separating pytest and pylint after seeing some of these PR's?

SchrodingersStruggle commented:
@kylemasc917 @avgagliardo Would either of you like to review this, seeing as the task you volunteered for is on topic?

laserlab commented:
I suggest @zbpetersbuf to review this so there is also a communication about data types. Since this data has to be read in by the preparation.py functions

laserlab commented:
> @kylemasc917 @avgagliardo Would either of you like to review this, seeing as the task you volunteered for is on topic?

Whoever reviews it check with @zbpetersbuf as well since the data types have to be compatible

laserlab commented:
> Yeah I see that your code has a 10/10 linting score so it deserves to pass the test. @laserlab do you think it's worth separating pytest and pylint after seeing some of these PR's?

Yes, I think separating it as in HW4 makes sense. You can check with @kylemasc917 @HeshamElsaman who did it like that

laserlab commented:
> Just confirming, I don't need unit tests for my python scripts for this PR, right?

No unit tests, but your json data is no processed it’s just copied as is. A more reasonable data format for a pandas dataframe would be a datetime as mentioned in class

haiderabbas007 commented:
I modified all the .json files by adding the following line of code in my `*.py` files and added the modified files to the directory:

`df_extracted['Timestamp'] = pd.to_datetime(df_extracted['Timestamp'], utc=True)
`

Are these acceptable now?

laserlab commented:
flask_monthly.json:

```json
[{"CO2 (ppm)":null,"Timestamp":null},{"CO2 (ppm)":null,"Timestamp":null},{"CO2 (ppm)":null,"Timestamp":null},{"CO2 (ppm)":null,"Timestamp":null},{"CO2 (ppm)":null,"Timestamp":null},{"CO2 (ppm)":null,"Timestamp":null},{"CO2 (ppm)":null,"Timestamp":null},{"CO2 (ppm)":null,"Timestamp":null},{"CO2 (ppm)":null,"Timestamp":null},{"CO2 (ppm)":null,"Timestamp":null},{"CO2 (ppm)":null,"Timestamp":null},{"CO2 (ppm)":null,"Timestamp":null},{"CO2 (ppm)":null,"Timestamp":null},{"CO2 (ppm)":null,"Timestamp":null},{"CO2 (ppm)":null,"Timestamp":null},{"CO2 (ppm)":null,"Timestamp":null},{"CO2
…
```

Seems not to work

haiderabbas007 commented:
nearly fixed, few min

laserlab commented:
Taking this as a reviewer since I already discussed it in class.

laserlab reviewed (changes requested):
Looks good so far, but please select one data set that is useful for determining the seasonal trend for Mount Lua (meaning contiguous and equidistant data) and only keep the json as the data file.

haiderabbas007 commented:
I have selected the flask sampling (monthly) dataset. The new repository as directed is found here 

https://github.com/haiderabbas007/CP1-24-HW5/tree/005c14c480ff29a09d0d21974917d933996305da/mauna-loa-data

@laserlab

laserlab commented:
@SchrodingersStruggle let’s merge this now so we have a data set

laserlab reviewed (approved):
Still issues, but we can fix those along the way. Let’s merge now

SchrodingersStruggle commented:
Okay sounds good, merging.

haiderabbas007 commented:
> Still issues, but we can fix those along the way. Let’s merge now

Thanks, out of curiosity, what issues need to be fixed?
